typing/oprint.ml

```diff
@@ -499,6 +526,8 @@ and print_simple_out_module_type ppf =
           fprintf ppf "@[<hv 2>sig@ %a@;<1 -2>end@]" print_out_signature sg
      end
   | Omty_alias id -> fprintf ppf "(module %a)" print_ident id
+  | Omty_functor _ as non_simple ->
+     fprintf ppf "(%a)" print_out_module_type non_simple
```

Comment:
This goes to the bottom because it is a catch-all case that may evolve to contain other non-simple constructors in the future.